src/mm/allocator.rs

```diff
@@ -11,14 +11,22 @@ use crate::mm::hole::{Hole, HoleList};
 use crate::mm::kernel_end_address;
 use crate::synch::spinlock::*;
 use core::alloc::{AllocError, GlobalAlloc, Layout};
+use core::cmp;
 use core::ops::Deref;
 use core::ptr::NonNull;
 use core::{mem, ptr};
 
 /// Size of the preallocated space for the Bootstrap Allocator.
 const BOOTSTRAP_HEAP_SIZE: usize = 4096;
 
+#[cfg(target_arch = "x86_64")]
+const CACHE_LINE_SIZE: usize = 128;
+#[cfg(not(target_arch = "x86_64"))]
+const CACHE_LINE_SIZE: usize = 64;
+
 /// A fixed size heap backed by a linked list of free memory blocks.
+#[cfg_attr(target_arch = "x86_64", repr(align(128)))]
+#[cfg_attr(not(target_arch = "x86_64"), repr(align(64)))]
```

Comment:
Same as before, why do you align the struct to 128 on x86?

Reply:
Also can you reuse the CACHE_LINE_SIZE constant here?

Reply:
I think that number in "align" must be a real number. You are not able to use a Rust constant.

Reply:
You are right, didn't know that this is not possible.